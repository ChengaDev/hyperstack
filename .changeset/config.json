{
  "$schema": "https://unpkg.com/@changesets/config@2.0.0/schema.json",
  "changelog": false,
  "commit": false,
  "linked": [],
  "access": "public",
  "baseBranch": "master",
  "updateInternalDependencies": "patch",
  "ignore": [],
  "fixed": [
    [
      "create-hyperstack",
      "template-app",
      "template-blank",
      "hyperstack",
      "@hyperstackjs/*"
    ]
  ]
}
