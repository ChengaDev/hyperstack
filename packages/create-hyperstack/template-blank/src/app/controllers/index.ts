// naming and imports: name and import as you see fit

const controllers: any[] = []
// controllers.push(NotesController)

export const AppController = {
  controllers,
}
