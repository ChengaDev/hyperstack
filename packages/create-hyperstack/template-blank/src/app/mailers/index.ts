// suggested mailer layout:
// mailer-name.ts
// mailer-name/
//  template-contents
//
// export your mailers here:
// export { MyMailer } from './my-mailer'

export {}
