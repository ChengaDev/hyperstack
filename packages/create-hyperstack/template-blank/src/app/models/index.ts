// export your models here
// export { MyModel } from './my-model'

export {}
