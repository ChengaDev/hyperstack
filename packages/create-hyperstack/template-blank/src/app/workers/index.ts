// export your workers here:
// e.g. export { MyWorker } from './my-worker'
export {}
