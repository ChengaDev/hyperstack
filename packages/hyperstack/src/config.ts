import path from 'path'
import L from 'lodash'
import createDebug from 'debug'
import type { Config } from '@hyperstackjs/typings'
import { getAppEnvironment } from './context'
import { requireModule } from './utils'
import type { ConfigPaths, Context } from './types'
const debug = createDebug('hyperstack:config')

export const resolveConfig = ({ exactPath, appPath }: ConfigPaths) => {
  if (exactPath) {
    return exactPath
  }
  if (!appPath) {
    throw new Error('app path is empty')
  }
  return path.join(appPath, `config/environments/${getAppEnvironment()}`)
}
export const loadConfig = async (paths: ConfigPaths) => {
  const resolved = resolveConfig(paths)

  const mod = requireModule(resolved)

  // configs are function providing a dict.
  // it's a function because merging modifies the provided dict, and we
  // don't want to fiddle with module cache in case we want live reloads or tests to work
  const config = (await mod()) as Config

  debug('config before overrides is %o', config)
  if (paths.configOverrides) {
    L.merge(config, paths.configOverrides)
  }
  return { config, resolvedConfigPath: resolved }
}

export const tieConfig = async (
  context: Context,
  { exactPath, configOverrides }: ConfigPaths = {}
) => {
  const appPath = context.store().app!.root
  const { config, resolvedConfigPath } = await loadConfig({
    configOverrides,
    exactPath,
    appPath,
  })
  debug('loaded from %o', resolvedConfigPath)
  debug('config is %o', config)
  debug('overrides %o', configOverrides)
  context.setConfig(config)
}
