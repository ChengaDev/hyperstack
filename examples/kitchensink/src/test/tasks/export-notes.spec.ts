import { test } from '@hyperstackjs/testing'
import ExportNotes from '../../lib/tasks/export-notes'
import { appContext } from '../../app'

import { root } from '../../config/settings'

const { tasks } = test(root)

const pid = /\|.+-.+-.+-.+-.+/

describe('tasks', () => {
  describe('export-notes', () => {
    tasks('should work', async (app) => {
      const { User, Note } = appContext.models()
      const user = await User.createWithPassword({
        username: 'evh@example.com',
        password: 'mypass-should-login',
        name: 'Eddie Van Halen',
      })
      await Note.createWithOwner(user, {
        content: 'interesting note',
        title: 'this is the title',
      })
      expect(
        (await ExportNotes.exec({ username: user.username }, app)).replace(
          pid,
          'test-redacted'
        )
      ).toMatchSnapshot()
    })
  })
})
