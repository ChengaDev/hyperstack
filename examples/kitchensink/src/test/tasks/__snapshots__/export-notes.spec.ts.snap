// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tasks export-notes should work 1`] = `
"content|title|pid
interesting notetest-redacted"
`;
